webserver/login/__init__.py

```diff
@@ -33,6 +34,21 @@ def load_user(user_id):
     else:
         return None
 
+@login_manager.request_loader
+def load_user(request):
+    key = request.headers.get("Authorization")
+    user = None
+    if key:
+        parts = key.split(" ")
+        if len(parts) == 2 and parts[0] == "Token":
+            user = db.user.get_by_api_key(parts[1])
+        else:
+            raise Unauthorized
+    if user:
+        return User.from_dbrow(user)
+    else:
+        return None
```

Comment:
I had a bit of a look at flask-login, it looks like it only works on the app level. There are a few things we could do here instead:

* Check in the `request_loader` the value of `request.path` and return `None` if we're not in the API blueprint (I don't like this)
* Set a flag that says we were authed by token (where? `flask.g`?) and check this flag in a decorator on non-api endpoints, denying access if it's set (probably replacing the flask `@login_required` decorator, but this means we modify non-api endpoints for the sake of api auth)
* Go back to doing auth in a specific decorator for the API (this means setting `flask_login.current_user` properly. Is that all we have to do to make a user appear logged in?)

I'm not sure we have to restrict web access to only session auth. As I understand we're already using session auth in the API anyway, when the website accesses an API endpoint with js. Do you have a specific reason why you don't like having the opposite too?
Theoretically the keys are supposed to be secure, and I don't think there's anything else that a user could do on the website which they can't do by the API. I don't mind having the possibility of token auth for the website, even if no one uses it.